src/equations/bbm_bbm_1d.jl

```diff
@@ -132,8 +132,9 @@ function create_cache(mesh, equations::BBMBBMEquations1D,
                       ::BoundaryConditionPeriodic,
                       RealT, uEltype)
     D = equations.D
-    invImD2 = inv(I - 1 / 6 * D^2 * Matrix(solver.D2))
-    return (invImD2 = invImD2,)
+    invImD2 = cholesky(Symmetric(I - 1 / 6 * D^2 * Matrix(solver.D2)))
+    tmp2 = Array{RealT}(undef, nnodes(mesh))
+    return (invImD2 = invImD2, tmp2 = tmp2)
```

Comment:
This is still a dense matrix, isn't it? Can you now use the rational operators or at least sparse matrices?

Reply:
IIRC, you were using `sparse` for the results posted at the top of this PR

Reply:
You mean I should define `invImD2 = cholesky(Symmetric(I - 1 / 6 * D^2 * sparse(solver.D2)))`?

Reply:
Yes

Reply:
This still gives this error:

>  I tried using `ldiv!(invImD2, deta)`, but this gave the following error message:
> ```julia
> ERROR: LoadError: MethodError: no method matching ldiv!(::SparseArrays.CHOLMOD.Factor{Float64, Int64}, ::SubArr
>```

Reply:
Yeah, this needs the 3-argument version of `ldiv!`

Reply:
Hm, like in [fe92848](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/108/commits/fe928487ed8a54f02ed91f2bb717a3c9dc66303a)? Still the same error.

Reply:
You would need another temporay array to store the result - it must not alias with the source like in `ldiv!(tmp1, factorization, tmp1)` but has to be independent, e.g., `ldiv!(tmp2, factorization, tmp1)`.

Reply:
Like [a9a9391](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/108/commits/a9a93914cb5e8859715cc49fca7549ee2ec82d89)? Again the same.

Reply:
Could you please show the full stacktrace of the error?

Reply:
Yes, looks like that's it 🤷 

Here is a code snippet that should work (where I have to update the system matrix in each step):
```julia
    system_matrix = Symmetric(...)
    if issparse(system_matrix)
        (; factorization) = parameters
        cholesky!(factorization, system_matrix)
        dv .= factorization \ tmp
    else
        factorization = cholesky!(system_matrix)
        ldiv!(dv, factorization, tmp)
    end
```

Reply:
Hm, this also doesn't seem to be ideal. I've also just seen that CI is complaining about the system matrix not being positive definite (I can't link to the line because of the GitHub Actions hiccups) and thus the Cholesky decomposition fails. But maybe it is easier (for now?) to go with an LU decomposition instead of a Cholesky decomposition?

Reply:
In [e68079e](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/108/commits/e68079ee1f4301890e398500bccdd386e072bc8c) I made a proposal for the LU decomposition, which works, has no allocations in the elliptic part and is reasonably fast, I think:
```julia
 ──────────────────────────────────────────────────────────────────────────────
        DispersiveSWE                 Time                    Allocations      
                             ───────────────────────   ────────────────────────
      Tot / % measured:           552ms /  97.8%            627MiB /  99.8%    

 Section             ncalls     time    %tot     avg     alloc    %tot      avg
 ──────────────────────────────────────────────────────────────────────────────
 rhs!                 16.8k    503ms   93.1%  30.0μs    610MiB   97.5%  37.3KiB
   dv hyperbolic      16.8k    132ms   24.4%  7.87μs    338MiB   54.0%  20.6KiB
   deta hyperbolic    16.8k    125ms   23.1%  7.43μs    270MiB   43.2%  16.5KiB
   deta elliptic      16.8k    117ms   21.7%  7.00μs     0.00B    0.0%    0.00B
   dv elliptic        16.8k    113ms   21.0%  6.77μs     0.00B    0.0%    0.00B
   ~rhs!~             16.8k   15.7ms    2.9%   934ns   2.56MiB    0.4%     160B
   source terms       16.8k    586μs    0.1%  34.9ns     0.00B    0.0%    0.00B
 analyze solution       280   37.0ms    6.9%   132μs   15.4MiB    2.5%  56.5KiB
 ──────────────────────────────────────────────────────────────────────────────
```

Reply:
Sounds good 👍